<launch>
    <arg name="tcp_ip" default="0.0.0.0"/>
    <arg name="tcp_port" default="10000"/>

    <node name="server_endpoint" pkg="ros_tcp_endpoint" type="default_server_endpoint.py" args="--wait" output="screen" respawn="true">
            <param name="tcp_ip" type="string" value="$(arg tcp_ip)"/>
            <param name="tcp_port" type="int" value="$(arg tcp_port)"/>
    </node>
    <node name="color_pipeline" pkg="ROS-Nursing-Robot-MQP-23-24" type="pipeline.py" output="screen" />
    <!--node pkg="tf" type="static_transform_publisher" name="adjust" args="0 0 0 1.57079 0 0 world adjust 100" / -->

    <node pkg="tf" type="static_transform_publisher" name="calibration_tag" args="-0.065 0.20 0 3.1415 0 0 adjust calibration_tag 100" />

    <node pkg="tf" type="static_transform_publisher" name="grey_cube_center" args="0 0 -0.026 0 0 0 grey_cube grey_cube_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="corn_can_center" args="0 0 -0.056 0 0 0 corn_can corn_can_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="bottle_2_center" args="0 0 -0.046 0 0 0 bottle_2 bottle_2_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="white_cup_center" args="0 0.0412 -0.04 -1.579079 0 0 white_cup white_cup_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="white_cup_bowl" args="0 0.065 0 -1.579079 0 0 white_cup white_cup_bowl 100" />
    <node pkg="tf" type="static_transform_publisher" name="blue_cup_center" args="0 0.065 -0.04 -1.57079 0 0 blue_cup blue_cup_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="blue_cup_handle" args="0 0.13 0 -1.57079 0 0 blue_cup blue_cup_handle 100" />

    <!-- <node pkg="tf" type="static_transform_publisher" name="gripper_link" args="0.5 0 0 0 0 0 gripper_link adjust 100" /> -->
    <node pkg="tf" type="static_transform_publisher" name="arm_origin" args="-0.11 -0.095 -0.107 -1.57079 0 0 robot_adjust arm_origin 100" />
    <node pkg="tf" type="static_transform_publisher" name="arm_relative" args="-0.105 0.100 -0.107 -1.57079 0 0 robot_adjust arm_relative 100" />
    <node pkg="tf" type="static_transform_publisher" name="base_link" args="-0.17 0 0 0 0 0 arm_origin base_link 100" />
    <node name="unity_publisher" pkg="ROS-Nursing-Robot-MQP-23-24" type="unity_publisher.py" output="screen" />
    <node name="intermediate_publisher" pkg="ROS-Nursing-Robot-MQP-23-24" type="intermediate_publisher.py" output="screen" />
    <node name="teleop" pkg="ROS-Nursing-Robot-MQP-23-24" type="teleop.py" output="screen" />
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ROS-Nursing-Robot-MQP-23-24)/rviz/rviz.rviz"/>
 </launch>
